sycl/include/sycl/ext/intel/experimental/esimd/memory.hpp

```diff
@@ -2759,13 +2759,13 @@ lsc_atomic_update(T *p, __ESIMD_NS::simd<Toffset, N> offsets,
   constexpr detail::lsc_data_order _Transposed =
       detail::lsc_data_order::nontranspose;
   using _MsgT = typename detail::lsc_expand_type<T>::type;
+  __ESIMD_NS::simd<_MsgT, N> _Msg_data = detail::lsc_format_input<_MsgT>(src0);
```

Comment:
- How did this code work without bitcast to uint32_t when T=float?
- We need to minimize the number of bit_casts. They are supposed to be harmless/NO-OP, but in fact may cause bad code-gen in GPU Jit.

Reply:
Yes, right, I missed the bit_cast in the code "before". Thank you.